Automattic/newspack-network
# feat(content-distribution): handle post thumbnail

Add support for syncing the post thumbnail when distributing content.

### Testing

1. Make sure you have a network setup with at least 1 hub and 1 node
2. On the hub, create a new post with a featured image, grab the post ID, and use the WP shell to distribute to the node:

```php
$post_id = 0; // Change to your post ID.
$site_urls = [ 'http://node1.local' ]; // Change to match your setup urls.
Newspack_Network\Content_Distribution::set_post_distribution( $post_id, $site_urls );
Newspack_Network\Content_Distribution::distribute_post( $post_id );
```

3. Visit the media library and confirm the post thumbnail was created
4. Confirm the post is added as a draft and the thumbnail is assigned
5. In the original post, change the featured image and save
6. Visit the media library in node1 and confirm that the previous featured image has been deleted and the new one was created
7. Confirm the new featured image was also updated in the post
8. In the original post, delete the featured image and save
9. Back in node1, confirm in the media library that the featured image has been deleted
10. Confirm in the linked post that no featured image is assigned
